### PR TITLE
Use jsonschema for validating the config file

### PR DESCRIPTION
**Pull request by jaredly**
## Summary:
I'm going to be changing the config file format quite a bit, and it'll be nice to have actual validation.

Issue: XXX-XXXX

## Test plan:
See tests

### CODE REVIEW

File: src/cli/config.js [hunk 1]
```diff
@@ -78,6 +74,9 @@ type SubConfig = {
     extends?: string,
 };
 
+const subSchema = {...jsonSchema, required: []};
+subSchema.properties = {...jsonSchema.properties, extends: {type: 'string'}};
```
Comment (kevinbarabash) [inline src/cli/config.js:78]: Does it make sense to allow `schemaFilePath` at all in sub schemas?

File: src/cli/config.js [hunk 2]
```diff
@@ -92,7 +91,7 @@ export const loadSubConfigFile = (configFile: string): SubConfig => {
             );
         }
     });
-    validateOptions(configFile, data.options ?? {});
+    validateOrThrow(data, subSchema);
```
Comment (kevinbarabash) [inline src/cli/config.js:94]: I don't understand how `extends` works.  It looks like we're reading the sub config, but not copying any fields from the base config.  Does that happen somewhere else?

File: src/cli/run.js [hunk 3]
```diff
@@ -46,6 +47,11 @@ const findGraphqlTagReferences = (root: string): Array<string> => {
 
 const [_, __, configFile, ...cliFiles] = process.argv;
 
+// eslint-disable-next-line flowtype-errors/uncovered
+const jsonSchema = JSON.parse(
+    fs.readFileSync(__dirname + '/config.json', 'utf8'),
+);
```
Comment (kevinbarabash) [inline src/cli/run.js:53]: It's too bad `jsonschema` doesn't provide a well-typed parse function.

File: src/cli/config.js [hunk 4]
```diff
@@ -4,6 +4,7 @@ import type {Schema} from '../types';
 import type {GraphQLSchema} from 'graphql/type/schema';
 
 import {schemaFromIntrospectionData} from '../schemaFromIntrospectionData';
+import jsonSchema from './schema.json';
```
Comment (kevinbarabash) [inline src/cli/config.js:7]: It's a little bit confusing having a variable named `jsonSchema` while we're using a module named `jsonschema`.  Maybe rename this variable to `schema` or `configSchema`.

**Review by kevinbarabash — APPROVED** [review]
Super cool!  I love that we're using JSON schema for something.  This is a perfect use case for it.  A few nits in the comments, but nothing blocking.  Thanks for adding this. 🎉

File: src/cli/config.js (hunk 2 above)
Comment (jaredly) [reply src/cli/config.js:94]: yeah so this stuff is all getting replaced in https://github.com/Khan/graphql-flow/pull/48
there will only be a single config file, no `extends`.

**Pull request merged**